tooling/minimize/src/main.rs

```diff
@@ -111,42 +117,88 @@ pub const DEFAULT_ARGS: &[&str] = &[
     "-Zub-checks=false",
 ];
 
-fn show_error(msg: &impl std::fmt::Display) -> ! {
+pub fn show_error(msg: &impl std::fmt::Display) -> ! {
     eprintln!("fatal error: {msg}");
     std::process::exit(101) // exit code needed to make ui_test happy
 }
 
+#[macro_export]
 macro_rules! show_error {
-    ($($tt:tt)*) => { crate::show_error(&format_args!($($tt)*)) };
+    ($($tt:tt)*) => {crate::show_error(&format_args!($($tt)*)) };
+}
+struct BeRustcCallbacks;
+
+impl rustc_driver::Callbacks for BeRustcCallbacks {}
+
+pub fn be_rustc(args: &mut Vec<String>) {
+    let use_panic_abort = args
+        .array_windows::<2>()
+        .any(|[first, second]| first == "--crate-name" && second == "panic_abort");
+
+    if use_panic_abort {
+        args.insert(1, "-Cpanic=abort".into());
+    }
+
+    let exit_code = rustc_driver::catch_with_exit_code(move || {
+        rustc_driver::run_compiler(args, &mut BeRustcCallbacks)
+    });
+
+    std::process::exit(exit_code);
 }
 
 fn main() {
```

Comment:
So here's what should happen in main:
- first check `MINIMIZE_BUILD_SYSROOT`
- if that is set to "only", do the sysroot build and exit
- if it is set to "off", do nothing for now
- otherwise, build the sysroot and remember where it was put

Then prepare the arguments:
- splice in DEFAULT_ARGS
- splice in the sysroot, if we build one above

Then continue with these arguments -- the rest of your logic looks good.